src/monster.cpp

```diff
@@ -2139,6 +2153,37 @@ void monster::die( Creature *nkiller )
     }
 }
 
+bool monster::mod_mech_power( int amt )
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    amt = -amt;
+    battery_item->ammo_consume( amt, pos() );
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    return true;
+}
+
+bool monster::check_mech_powered()
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    itype_id mech_bat = itype_id( type->mech_battery );
+    const itype &type = *item::find_type( mech_bat );
```

Comment:
Why don't you use `item::type` of the `battery_item` directly? According to the constructor, `battery_item` should contain an item of type `type->mech_battery`.
```suggestion
    const itype &type = *battery_item->type;
```